xlive/H2MOD/Modules/Stats/StatsHandler.cpp

```diff
@@ -0,0 +1,686 @@
+		//LOG_ERROR_GAME(curl_easy_strerror(result));
+		return -1;
+	}
+	else
+	{
+		int response_code;
+		curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response_code);
+		LOG_INFO_GAME(std::to_string(response_code));
+		return response_code;
+	}
+	return -1;
+}
+
+
+//Really should make structs for these but I don't want to take the time to learn how.
+static const int baseOffset  = 0x4dc722;
+static const int rtPCROffset = 0x4DD1EE;
+static const int PCROffset   = 0x49F6B0;
+
+char* StatsHandler::buildJSON()
+{
+	typedef rapidjson::GenericDocument<rapidjson::UTF16<>> WDocument;
+	typedef rapidjson::GenericValue<rapidjson::UTF16<>> WValue;
+	WDocument document;
+	WValue value;
+	document.SetObject();
+
+
+	//Inital Properties
+	rapidjson::Document::AllocatorType& allocator = document.GetAllocator();
+	auto checksum = getChecksum();
+	auto wchecksum = std::wstring(checksum.begin(), checksum.end());
+	value.SetString(wchecksum.c_str(), allocator);
+	document.AddMember(L"PlaylistChecksum", value, allocator);
+	auto Scenario = h2mod->GetAddress<wchar_t*>(0, 0x4DC504);
+
+	std::wstring ScenarioPath = std::wstring(Scenario);
+	std::wstring Scen(ScenarioPath.substr(ScenarioPath.rfind('\\') + 1));
+
+	value.SetString(Scen.c_str(), allocator);
+	document.AddMember(L"Scenario", value, allocator);
+
+	//Build the Variant Object
+	WValue Variant(rapidjson::kObjectType);
+	auto VariantName = h2mod->GetAddress<wchar_t*>(0, 0x4DC3D4);
+	value.SetString(VariantName, allocator);
+	Variant.AddMember(L"Name", value, allocator);
+	BYTE VariantType = *h2mod->GetAddress<BYTE*>(0, 0x4dc414);
+	Variant.AddMember(L"Type", VariantType, allocator);
+	
+	WValue VariantSettings(rapidjson::kObjectType);
+	BYTE TeamPlay = *h2mod->GetAddress<BYTE*>(0, 0x992880);
+	value.SetString(IntToWString<BYTE>(TeamPlay, std::dec).c_str(), allocator);
+	VariantSettings.AddMember(L"Team Play", value, allocator);
+	Variant.AddMember(L"Settings", VariantSettings, allocator);
+	document.AddMember(L"Variant", Variant, allocator);
+	
+	//Build the Server Object
+	WValue Server(rapidjson::kObjectType);
+	auto ServerXUID = *h2mod->GetAddress<::XUID*>(0, 0x52FC50);
+	auto sXUID = IntToWString<::XUID>(ServerXUID, std::dec).c_str();
+	value.SetString(sXUID, allocator);
+	Server.AddMember(L"XUID", value, allocator);
+	//h2mod->get_local_player_name(0)
+	auto ServerName = h2mod->GetAddress<wchar_t*>(0, 0x52FC88);
+	value.SetString(ServerName, allocator);
+	Server.AddMember(L"Name", value, allocator);
+	auto IP = NetworkSession::getLocalNetworkAddress();
+	auto IPString = std::string(inet_ntoa(IP));
+	auto wIPString = std::wstring(IPString.begin(), IPString.end());
+	value.SetString(wIPString.c_str(), allocator);
+	Server.AddMember(L"IP", value, allocator);
+	document.AddMember(L"Server", Server, allocator);
+
+	//Players
+	int playerCount = 0;
+	WValue Players(rapidjson::kArrayType);
+	for(auto i = 0; i < 16; i++)
+	{
+		int calcBaseOffset  = baseOffset + (i * 0x94);
```

Comment:
please break up this method into smaller methods for future development sanity.,...